### PR TITLE
Replace Pipe with Callable in Language

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

Replace `Pipe` with `Callable` in `Language`.

### Types of change

?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
Well, this reminded me why we're using `Pipe` here. It looks like mypy may support `hasattr` soon, so I think once that is supported, it would make sense to switch to the correct type `Callable` instead of `Pipe` in `Language.

File: spacy/language.py [hunk 1]
```diff
@@ -180,7 +176,7 @@ def __init__(
         self.vocab: Vocab = vocab
         if self.lang is None:
             self.lang = self.vocab.lang
-        self._components: List[Tuple[str, "Pipe"]] = []
+        self._components: List[Tuple[str, Callable]] = []
```
Comment (rmitsch) [inline spacy/language.py:179]: The `Callable` would have to return a `Doc`, right? Should we type this as `Callable[..., Doc]` then?

Comment (adrianeboyd) [reply spacy/language.py:179]: That's a good point. Then it would be more accurate as: `Callable[[Doc], Doc]`

**Comment by robookwus** [comment]
Hey I'd like to help with the docs, or create my own version, how do I proceed. I emailed the developers and it was OK'd. Still a GitHub greenhorn but enjoy technical writing. Thx

**Comment by adrianeboyd** [comment]
@robookwus: Welcome! This particular thread is for a pull request that's not related to the docs.

Instead, please open a new issue (https://github.com/explosion/spaCy/issues/new?assignees=&labels=&template=02_docs.md) with more details about what you're interested in working on?

**Comment by adrianeboyd** [comment]
Replaced by #11803.

**Pull request closed without merging**